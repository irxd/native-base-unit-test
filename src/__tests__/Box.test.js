import React from 'react';
import { render } from '@testing-library/react-native';
import {
  NativeBaseProvider, Box
} from 'native-base';

describe('<Box />', () => {
  const { queryByTestId } = render(
    <NativeBaseProvider
      initialWindowMetrics={{
        frame: { x: 0, y: 0, width: 0, height: 0 },
        insets: { top: 0, left: 0, right: 0, bottom: 0 },
      }}
    >
      <Box testID="my-box">Test Box</Box>
    </NativeBaseProvider>
  )
  it('should render box component properly', () => {
    expect(queryByTestId('my-box')).not.toBeNull()
  });

})