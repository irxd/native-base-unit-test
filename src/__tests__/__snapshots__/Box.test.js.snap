// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> should render box props properly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderColor": "#000000",
        "marginBottom": 16,
      }
    }
    testID="my-box"
  >
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#000000",
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 21,
          "textDecorationLine": undefined,
        }
      }
    >
      Test Box
    </Text>
  </View>
</RNCSafeAreaProvider>
`;
